provisioning/transport/amqp/src/AmqpTransportHandler.cs

```diff
@@ -0,0 +1,145 @@
+
+namespace Microsoft.Azure.Devices.Provisioning.Client.Transport.Amqp
+{
+    internal static class AmqpTransportHandler
+    {
+        public const string clientVersion = "test-client-version";
+        public const string apiVersion = "2017-08-31-preview";
+
+        private static async Task<AmqpClientConnection> CreateConnection(Uri uri, ProvisioningSecurityClientSasToken securityClient, string linkendpoint)
+        {
+            AmqpSettings settings = await CreateAmqpSettings(securityClient, linkendpoint).ConfigureAwait(false);
+            return new AmqpClientConnection(uri, settings);
+        }
+
+        private static async Task<AmqpSettings> CreateAmqpSettings(ProvisioningSecurityClientSasToken securityClient, string linkendpoint)
+        {
+            var settings = new AmqpSettings();
+            byte[] ekBuffer = await securityClient.GetEndorsementKeyAsync().ConfigureAwait(false);
+            byte[] srkBuffer = await securityClient.GetStorageRootKeyAsync().ConfigureAwait(false);
+
+            string ek = Convert.ToBase64String(ekBuffer);
+            string srk = Convert.ToBase64String(srkBuffer);
+
+            TpmNetworkCredential credential = new TpmNetworkCredential(ek, srk);
+
+                var saslProvider = new SaslTransportProvider();
+                saslProvider.Versions.Add(AmqpConstants.DefaultProtocolVersion);
+                settings.TransportProviders.Add(saslProvider);
+
+            SaslTpmHandler tpmHandler = new SaslTpmHandler(ekBuffer, srkBuffer, linkendpoint, securityClient);
+            saslProvider.AddHandler(tpmHandler);
+
+            var amqpProvider = new AmqpTransportProvider();
+            amqpProvider.Versions.Add(AmqpConstants.DefaultProtocolVersion);
+            settings.TransportProviders.Add(amqpProvider);
+            return settings;
+        }
+
+        internal static async Task<AmqpClientConnection> CreateAmqpCloudConnectionAsync(
+            string deviceEndpoint, 
+            X509Certificate2 clientCert, 
+            string linkEndpoint, 
+            bool useWebSocket,
+            ProvisioningSecurityClientSasToken securityClient)
+        {
+            AmqpClientConnection amqpCloudConnection;
+            if (useWebSocket)
+            {
+                // TODO: enable WS
+                amqpCloudConnection = await CreateConnection(
+                    new Uri(WebSocketConstants.Scheme + deviceEndpoint + ":" + WebSocketConstants.Port),
+                    securityClient,
+                    linkEndpoint).ConfigureAwait(false);
+        }
+        else
+            {
+                amqpCloudConnection = await CreateConnection(
+                    new Uri("amqps://" + deviceEndpoint + ":" + AmqpConstants.DefaultSecurePort), 
+                    securityClient, 
+                    linkEndpoint).ConfigureAwait(false);
+            }
+
+            await amqpCloudConnection.OpenAsync(TimeSpan.FromMinutes(1), useWebSocket, clientCert).ConfigureAwait(false);
```

Comment:
Please extract the TimeSpan as a const. We should consider exposing these timings on the AmqpTransportHandler public API.